src/compile.coffee

```diff
@@ -1,45 +1,50 @@
-
 clone = require('clone')
 caseless = require('caseless')
 
 {child, children, parent, content} = require('./refract')
 validateParameters = require('./validate-parameters')
 detectTransactionExamples = require('./detect-transaction-examples')
 expandUriTemplateWithParameters = require('./expand-uri-template-with-parameters')
-apiElementsToJson = require('./api-elements-to-json')
+apiElementsToRefract = require('./api-elements-to-refract')
 
 
 compile = (mediaType, apiElements, filename) ->
-  parseResult = apiElementsToJson(apiElements)
-
   transactions = []
   errors = []
   warnings = []
 
-  component = 'apiDescriptionParser'
-  for annotation in children(parseResult, {element: 'annotation'})
-    group = if annotation.meta.classes[0] is 'warning' then warnings else errors
+  apiElements.annotations.forEach((annotation) ->
```

Comment:
I guess this is more of a personal preference, I find it easier to reason about and read when there is less mutation.

```js
// X? I am not sure what this should be called
annotationToX = (annotation) => {
  return {
    component: 'apiDescriptionParser'
    code: annotation.code?.toValue()
     message: annotation.toValue()
     location: annotation.sourceMapValue
  }
}

const errors = apiElements.annotations
  .filter(annotation => annotation.classes.contains('error'))
  .map(annotationToX)

const warnings = apiElements.annotations
  .filter(annotation => annotation.classes.contains('warning'))
  .map(annotationToX)
```